.pre-commit-config.yaml

```diff
@@ -50,7 +50,7 @@ repos:
         pass_filenames: false
       - id: mypy
         name: mypy
-        entry: mypy --python-executable $(pdm info --python)
+        entry: bash -c "mypy --python-executable $(pdm info --python) $@" --
```

Comment:
Does it work on win? I don't confirm the bash is always available on win.

Reply:
Good question, I never tried it  - -

Reply:
I mean it seems not to support pre-commit on win. And the pre-commit is absent in the Makefile or in CI. So it can't verify its working.

Reply:
Yes, I will try it when Win is available. It should fail when I try to push the code, I also feel a little confused when it does not take effect then.

Reply:
After some investigation, there is no easy solution for cross-platform command substitution or environment variable substitution. The maintainer of pre-commit also suggests [the `bash -c` and `env` approach](https://github.com/pre-commit/pre-commit/issues/758#issuecomment-505935221). :(